server/src/main/java/org/apache/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -259,7 +259,8 @@ public Pair<String, SupervisorSpec> map(int index, ResultSet r, StatementContext
                           );
                         }
                         catch (IOException e) {
-                          throw new RuntimeException(e);
+                          log.error(e, "Exception mapping SupervisorSpec for spec_id: [%s]", r.getString("spec_id"));
```

Comment:
Nit:
```suggestion
                          log.error(e, "Could not map json payload to a SupervisorSpec for spec_id [%s]", r.getString("spec_id"));
```

Reply:
The exception stack trace would include only snippets of the json payload, so we should probably include the json payload in the log message itself.

Reply:
Actually, I am not logging it since we have the payload as bytes.

Reply:
Makes sense, totally glossed over that.